File: psra/service/src/main/java/by/arhor/psra/dto/PhotoDto.java
package by.arhor.psra.dto;

import java.util.Set;
import java.util.Objects;
import java.util.StringJoiner;

public class PhotoDto extends Dto {

  private String name;
  private String description;
  private String path;
  private Set<String> tags;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  public Set<String> getTags() {
    return tags;
  }

  public void setTags(Set<String> tags) {
    this.tags = tags;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    if (!super.equals(o)) {
      return false;
    }
    PhotoDto photoDto = (PhotoDto) o;
    return Objects.equals(name, photoDto.name)
        && Objects.equals(description, photoDto.description)
        && Objects.equals(path, photoDto.path)
        && Objects.equals(tags, photoDto.tags);
  }

  @Override
  public int hashCode() {
    return Objects.hash(super.hashCode(), name, description, path, tags);
  }

  @Override
  public String toString() {
    return new StringJoiner(", ", PhotoDto.class.getSimpleName() + "[", "]")
        .add("name='" + name + "'")
        .add("description='" + description + "'")
        .add("path='" + path + "'")
        .add("tags=" + tags)
        .toString();
  }
}

File: psra/service/src/main/scala/by/arhor/psra/dto/PhotoDto.scala
package by.arhor.psra.dto

import java.util
import java.util.Objects

import scala.beans.BeanProperty

class PhotoDto extends Dto {

  @BeanProperty var name: String = _
  @BeanProperty var description: String = _
  @BeanProperty var path: String = _
  @BeanProperty var tags: util.Set[String] = _

  override def equals(obj: Any): Boolean = {
    if (super.equals(obj) && getClass == obj.getClass) {
      val photo = obj.asInstanceOf[PhotoDto]
      name == photo.name &&
      description == photo.description &&
      path == photo.path &&
      tags == photo.tags
    } else {
      false
    }
  }

  override def hashCode(): Int = super.hashCode() +
    Objects.hash(getName, getDescription, getPath) +
    (if (getTags != null) getTags.stream.mapToInt(_.hashCode).sum else 0)

}
